informer/informer.go

```diff
@@ -339,6 +350,52 @@ func (i *Informer) isCachedFilled() bool {
 	return false
 }
 
+func (i *Informer) newWatcher(ctx context.Context) (watch.Interface, error) {
```

Comment:
We have retries and timeouts combined so we definitely get a working watcher out of it. This seems to be necessary quite often.